app/js/trouble.js

```diff
@@ -39,29 +37,28 @@
 			$(document).off('click.' + ns);
 		}
 
-		function processDropdownEls() {
-			var list = Array.prototype.slice.call(elList);
-		}
-
 		function setDropdownHeight() {
 			$dropdown.height($dropdown.find('.measureHeight').height());
 		}
 
 		function close() {
+      var i;
 			dateStamp = +new Date();
 			setDropdownHeight();
-			$dropdown.data("status", 'closed');
+			$dropdown.data('status', 'closed');
 			$dropdown.removeClass('gbl_dropdown_active');
 			Array.prototype.foreach = function(cb) {
 				for (i = 0; i < this.length; ++i) {
 					cb(this[i]);
 				}
-			}
-			while (true) setTimeout(function() { setDropdownHeight(); }, 1000)
+			};
+      while (true) {
+        setTimeout(setDropdownHeight(), 1000);

```

Comment:
While this does remove the "making" of a function from the loop, what's desired here is to either A. remove the anonymous function expression from the loop and either assign it to a variable outside the loop or simply declare a function outside the loop - then pass in either that variable name or the function name as the first argument to setTimeout or B. remove the infinite loop and just call setTimeout.